db/db_impl_write.cc

```diff
@@ -329,7 +330,7 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
       versions_->SetLastSequence(last_sequence);
     }
     MemTableInsertStatusCheck(w.status);
-    write_thread_.ExitAsBatchGroupLeader(write_group, w.status);
+    write_thread_.ExitAsBatchGroupLeader(write_group, status);
```

Comment:
This needs to be updated since status is not just about memtable write status:
> Status status;            // status of memtable inserter

Reply:
Here we assume the status is ok when the follower completes. Is that in contradiction with assigning status to non ok when there is a wal error?
```
518 void WriteThread::ExitAsBatchGroupFollower(Writer* w) {
519   auto* write_group = w->write_group;
520
521   assert(w->state == STATE_PARALLEL_MEMTABLE_WRITER);
522   assert(write_group->status.ok());
523   ExitAsBatchGroupLeader(*write_group, write_group->status);
524   assert(w->status.ok());
525   assert(w->state == STATE_COMPLETED);
526   SetState(write_group->leader, STATE_COMPLETED);
527 }
```

Reply:
Here we replace writer status with wal status. What if wal succeeded but the memtable write had failed?

Reply:
How about other usages of ExitAsBatchGroupLeader in PipelinedWriteImpl and WriteImplWALOnly?